systemtest/src/test/java/io/strimzi/systemtest/security/oauth/OauthAuthorizationST.java

```diff
@@ -260,23 +270,171 @@ void testSuperUserWithOauthAuthorization() {
+
+        Map<String, String> config = new HashMap<>();
+        config.put("resources", "[\"Topic:x-*\"]");
+        config.put("scopes", "[\"Describe\"]");
+        config.put("applyPolicies", "[\"Dev Team A\"]");
+
+        newDevAPolicy.put("config", config);
+
+        LOGGER.info("Changing the Dev Team A policy for topics starting with x- and checking that job will not be successful");
+        KeycloakUtils.updatePolicyOfRealmClient(baseUri, token, newDevAPolicy, TEST_REALM, kafkaClientId);
+        assertThrows(WaitException.class, () -> ClientUtils.waitForClientSuccess(TEAM_A_PRODUCER_NAME, NAMESPACE, MESSAGE_COUNT));
+
+        JobUtils.deleteJobWithWait(NAMESPACE, TEAM_A_PRODUCER_NAME);
+
+        LOGGER.info("Changing back to the original settings and checking, if the producer will be successful");
+
+        config.put("scopes", "[\"Describe\",\"Write\"]");
+        newDevAPolicy.put("config", config);
+
+        KeycloakUtils.updatePolicyOfRealmClient(baseUri, token, newDevAPolicy, TEST_REALM, kafkaClientId);
+        teamAOauthClientJob = teamAOauthClientJob.toBuilder()
+            .withDelayMs(1000)
+            .build();
+
+        teamAOauthClientJob.producerStrimziOauthTls(CLUSTER_NAME).done();
+        ClientUtils.waitForClientSuccess(TEAM_A_PRODUCER_NAME, NAMESPACE, MESSAGE_COUNT);
+        JobUtils.deleteJobWithWait(NAMESPACE, TEAM_A_PRODUCER_NAME);
+
+        LOGGER.info("Changing configuration of Kafka back to it's original form");
+        KafkaResource.replaceKafkaResource(CLUSTER_NAME, kafka -> {
+            kafka.getSpec().getKafka().setListeners(new ArrayOrObjectKafkaListenersBuilder()
+                .addNewGenericKafkaListener()
+                    .withName("tls")
+                    .withPort(9093)
+                    .withType(KafkaListenerType.INTERNAL)
+                    .withTls(true)
+                    .withNewKafkaListenerAuthenticationOAuth()
+                        .withValidIssuerUri(keycloakInstance.getValidIssuerUri())
+                        .withJwksExpirySeconds(keycloakInstance.getJwksExpireSeconds())
+                        .withJwksRefreshSeconds(keycloakInstance.getJwksRefreshSeconds())
+                        .withJwksEndpointUri(keycloakInstance.getJwksEndpointUri())
+                        .withUserNameClaim(keycloakInstance.getUserNameClaim())
+                        .withTlsTrustedCertificates(
+                            new CertSecretSourceBuilder()
+                                .withSecretName(KeycloakInstance.KEYCLOAK_SECRET_NAME)
+                                .withCertificate(KeycloakInstance.KEYCLOAK_SECRET_CERT)
+                                .build())
+                        .withDisableTlsHostnameVerification(true)
+                    .endKafkaListenerAuthenticationOAuth()
+                .endGenericKafkaListener()
+                .build());
+        });
+
+        KafkaUtils.waitForKafkaReady(CLUSTER_NAME);
```

Comment:
Why do we need to do this? Looks like the tests behind it anyway don't do anything.

Reply:
That was just to return the changes to it's original state, so the next test will not be affected.

Reply:
Sorry, but this code is so hard to read I can't figure out what it's supposed to do, what it is testing, why it is testing it this way ... Also so much is going on there in the details that opens all sorts of questions.

Why don't you add a javadoc and describe the strategy of this test, what it is testing, and how. And I'd also refactor the code so there's less code in the main test method calling out to individual tasks performed in helper methods.

In the test you seem to be configuring the authentication, but I don't see you configure the authorization. Therefore I would assume that the combination of oauth authentication with reauthentication enabled, and without any authorization is in effect. But then in the middle of the test you change permissions for the team-a-producer client, which would have zero effect when keycloak authorization is not used. And how is changing permissions related to testing reauthentication in the first place?

Reply:
Discussed offline, gonna add javadoc for the test. Any other comments were not mentioned on chat.

Reply:
@scholzj about your comment -> do you think that the block is unnecessary? I'm just thinking that when we delete it now and someone will add test which will be affected by it, he will be finding where the problem is.

Reply:
Well, first of all I think tests relying on an order in which they are executed are not great. And when you change things there and back like here it will not safe time either. But that is probably for another PR. 

Second, until the followup tests are added, this is just wasting time. And you do not even know what kind of settings they will need. So maybe you can just leave some comment or something for whoever might add these tests one day and remove this, or?

Reply:
Yes, definitely. Thanks :)